### PR TITLE
Move tests for `integration-tests` from `/tests` to `/src/test`

### PR DESCRIPTION
**Pull request by pmnoxx**
Move tests for `integration-tests` from `/tests` to `/src/tests`.

As part of #5371

### CODE REVIEW

**Comment by pmnoxx** [comment]
http://nayduck.near.org/#/run/2257

**Comment by pmnoxx** [comment]
Running nayduck on expensive tests only: http://nayduck.near.org/#/run/2260

**Comment by mina86** [comment]
We need:

    sed -i 's/\(expensive integration-tests\) [^ ]\+ /\1 integration_tests tests::/' nightly/expensive.txt 

I believe.

File: integration-tests/src/lib.rs [hunk 1]
```diff
@@ -2,4 +2,6 @@ pub mod genesis_helpers;
 pub mod node;
 pub mod runtime_utils;
 pub mod test_helpers;
+#[cfg(test)]
+mod tests;
 pub mod user;
```
Comment (matklad) [inline integration-tests/src/lib.rs:7]: Odd ordering of modules, I'd put `#[cfg(test)] mod tests` into a separate block separated by a blank line.

Comment (pmnoxx) [reply integration-tests/src/lib.rs:7]: @matklad I'm waiting for https://github.com/near/nearcore/pull/5781

Comment (pmnoxx) [reply integration-tests/src/lib.rs:7]: I want to move network related types to `config.rs`.

**Review by chefsale — APPROVED** [review]

**Review by mina86 — APPROVED** [review]

**Pull request merged**